File: microservices/person/sqlite/factory.go
package sqlite

import (
	"database/sql"
	"log"
	"os"

	_ "github.com/mattn/go-sqlite3"
)

func GetConnection() *sql.DB {
	os.Remove("./rtdd.db")

	db, err := sql.Open("sqlite3", "./rtdd.db")

	if err != nil {
		log.Fatal(err)
	}

	return db
}

func ConfigConnection(db *sql.DB) {
	tablePerson := `
		CREATE TABLE person (
			id integer not null primary key autoincrement,
			name text,
			age integer
		);


		INSERT INTO person(name, age) VALUES ('Vinicius', 24);
		INSERT INTO person(name, age) VALUES ('Kevin', 23);
	`

	_, err := db.Exec(tablePerson)

	if err != nil {
		log.Fatal(err)
	}
}

File: microservices/product/src/adapter/sqlite/factory.ts
import sqlite3 from 'sqlite3'

const DBSOURCE = 'product.db'
const SQL_ITENS_CREATE = `
    CREATE TABLE product (
        id INTEGER PRIMARY KEY AUTOINCREMENT,
        name TEXT,
        price float
    );`

const INSERTED = "INSERT INTO product(name, price) VALUES ('Arroz', 10.25);"
const database = new sqlite3.Database(DBSOURCE, (err) => {
    if (err) {
        console.error(err.message)
        throw err
    } else {
        database.run(SQL_ITENS_CREATE, (_) => { })

        database.run(INSERTED, (_) => { })
    }
})

export { database }